### PR TITLE
Document Beads adoption boundaries and lessons

### LINKED ISSUE #618: Publish adoption guidance and feed Beads API lessons forward
Guidance and contract-feedback slice after real adoption work lands.

Acceptance criteria:
- Document when current code should depend directly on atelier.lib.beads, when tests should use atelier.testing.beads, and when work should wait for at-njpt4 instead.
- Record the concrete lessons, sharp edges, and follow-on API gaps discovered by the earlier adoption changesets.
- Update contributor guidance so future changes do not spread direct Beads usage into the wrong layers.
- Keep any cleanup strictly limited to guidance, contract feedback, and small adoption-adjacent simplifications.

Parent bead: at-k0ako

Local bead: at-k0ako.3

### PR DESCRIPTION
**Pull request by shaug**
# Summary

- publish contributor guidance for when to use `atelier.lib.beads`, when tests should use `atelier.testing.beads`, and when work should wait for the future store layer
- record the concrete lessons and current API/testing gaps surfaced by the recent low-level Beads adoption work

# Changes

- add `docs/beads-adoption-guide.md` with decision rules, anti-guidance, adoption lessons, and follow-on work
- cross-link the guide from the README and the existing Beads contract/testing docs
- add a docs regression test that keeps the published adoption boundary visible

# Testing

- `bash scripts/supported-python.sh run pytest tests/atelier/lib/test_beads_contract.py`
- `just format`
- `just lint`
- `just test`

# Tickets

- Fixes #618

# Risks / Rollout

- docs-only change; no runtime behavior changes

# Notes

- PR scope is measured against `main`; the stale epic root branch predates already-merged prerequisite changesets and would over-count unrelated work

### CODE REVIEW

File: docs/beads-adoption-guide.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+Use `atelier.lib.beads` directly when all of the following are true:
+
+- The code is a low-level adapter or helper that already owns Beads transport
+  concerns such as `BEADS_DIR`, repo cwd, or `--readonly` wiring.
+- The code benefits from typed request/response models more than it benefits
+  from a higher-level Atelier-owned store interface.
+- The seam is still close to raw issue semantics, not planner/worker business
+  policy.
+
+Use `atelier.testing.beads` by default when a test exercises Beads semantics
```
Comment (shaug) [inline docs/beads-adoption-guide.md]: This is still a bit too broad for `at-k0ako.3` relative to the projects current state. Success here is guidance that matches the adoption boundary we have *today*, not the one we expect after more migration. What done looks like: narrow this so `atelier.testing.beads` is recommended when the code under test already depends on `atelier.lib.beads`, and avoid implying that broader planner/worker unit-service suites are generally ready for in-memory backend use yet. Risk if left as-is: the docs will teach test-only in-memory adoption ahead of the production seam, which is the exact mismatch we just had to correct in `at-k0ako.1`.

Comment (shaug) [reply docs/beads-adoption-guide.md]: Endpoint probe for inline reply path

Comment (shaug) [reply docs/beads-adoption-guide.md]: Narrowed this to the current production seam.

- `docs/beads-adoption-guide.md` now says `atelier.testing.beads` applies only when the code under test already depends on `atelier.lib.beads`.
- `docs/in-memory-beads-testing-guide.md` now treats the migrated planner/worker suites as current adopted exceptions, not a general recommendation for broader unit-service coverage.
- `tests/atelier/lib/test_beads_contract.py` now locks in that narrower wording so the published guidance does not drift back toward test-only in-memory adoption ahead of the runtime boundary.
